pkg/azuredisk/azuredisk.go

```diff
@@ -126,3 +130,22 @@ func getResourceGroupFromDiskURI(diskURI string) (string, error) {
 	}
 	return fields[4], nil
 }
+
+func (d *Driver) checkDiskExists(ctx context.Context, diskURI string) error {
+	diskName, err := getDiskName(diskURI)
+	if err != nil {
+		return status.Error(codes.NotFound, fmt.Sprintf("Volume not found, failed with error: %v", err))
```

Comment:
`checkDiskExists` func should only `return err`, and then in `ControllerPublishVolume` func, you could wrap the error like this: `return status.Error(codes.NotFound, fmt.Sprintf("Volume not found, failed with error: %v", err)) `, because there are other functions that may invode this `checkDiskExists` and you could not know which error it should return in advance.